readthedocs/rtd_tests/tests/test_api.py

```diff
@@ -1137,6 +1231,83 @@ def test_gitlab_invalid_webhook(self, trigger_build):
         self.assertEqual(resp.status_code, 200)
         self.assertEqual(resp.data['detail'], 'Unhandled webhook event')
 
+    def test_gitlab_invalid_payload(self, trigger_build):
+        client = APIClient()
+        wrong_secret = '1234'
+        integration = Integration.objects.create(
+            project=self.project,
+            integration_type=Integration.GITLAB_WEBHOOK,
+            secret=get_secret(),
+        )
+        self.assertNotEqual(integration.secret, wrong_secret)
+        headers = {
+            GITLAB_TOKEN_HEADER: wrong_secret,
+        }
+        resp = client.post(
+            '/api/v2/webhook/gitlab/{}/'.format(self.project.slug),
+            self.gitlab_payload,
+            format='json',
+            **headers
+        )
+        self.assertEqual(resp.status_code, 403)
+        self.assertEqual(resp.data['detail'], 'Payload not valid')
+
+    def test_gitlab_valid_payload(self, trigger_build):
+        client = APIClient()
+        integration = Integration.objects.create(
+            project=self.project,
+            integration_type=Integration.GITLAB_WEBHOOK,
+            secret=get_secret(),
+        )
+        headers = {
+            GITLAB_TOKEN_HEADER: integration.secret,
+        }
+        resp = client.post(
+            '/api/v2/webhook/gitlab/{}/'.format(self.project.slug),
+            {'object_kind': 'pull_request'},
+            format='json',
+            **headers
+        )
+        self.assertEqual(resp.status_code, 200)
+
+    def test_gitlab_empty_token(self, trigger_build):
+        client = APIClient()
+        integration = Integration.objects.create(
+            project=self.project,
+            integration_type=Integration.GITLAB_WEBHOOK,
+            secret=get_secret(),
+        )
+        headers = {
+            GITLAB_TOKEN_HEADER: '',
+        }
+        resp = client.post(
+            '/api/v2/webhook/gitlab/{}/'.format(self.project.slug),
+            {'object_kind': 'pull_request'},
+            format='json',
+            **headers
+        )
+        self.assertEqual(resp.status_code, 403)
+        self.assertEqual(resp.data['detail'], 'Payload not valid')
+
+    def test_gitlab_skip_token_validation(self, trigger_build):
+        client = APIClient()
+        integration = Integration.objects.create(
+            project=self.project,
+            integration_type=Integration.GITLAB_WEBHOOK,
+            secret='',
+        )
+        self.assertFalse(integration.secret)
+        headers = {
+            GITLAB_TOKEN_HEADER: 'skipped',
```

Comment:
Same here, I'm assuming this is testing for the current implementation of no secret

Reply:
Yes, it's